ospd_openvas/daemon.py

```diff
@@ -1116,6 +1116,38 @@ def report_openvas_timestamp_scan_host(
             )
             return
 
+    def openvas_process_is_alive(
+        self, kbdb: BaseDB, ovas_pid: str, openvas_scan_id: str
+    ) -> bool:
+        parent_exists = True
+        parent = None
+        try:
+            parent = psutil.Process(int(ovas_pid))
+        except psutil.NoSuchProcess:
+            logger.debug('Process with pid %s already stopped', ovas_pid)
+            parent_exists = False
+        except TypeError:
+            logger.debug(
+                'Scan with ID %s never started and stopped unexpectedly',
+                openvas_scan_id,
+            )
+            parent_exists = False
+
+        is_zombie = False
+        if parent and parent.status() == psutil.STATUS_ZOMBIE:
+            is_zombie = True
+
+        if (not parent_exists or is_zombie) and kbdb:
+            if kbdb and kbdb.scan_is_stopped(openvas_scan_id):
+                return True
+            kbdb.stop_scan(openvas_scan_id)
+            for scan_db in kbdb.get_scan_databases():
+                self.main_db.release_database(scan_db)
+            self.main_db.release_database(kbdb)
```

Comment:
This doesn't fit to a method that checks something and returns a value. If a method has a side effect (that means it does change some data) it should be clearly visible from the method name and signature. In this case i would move this code to another method and call this method if is_openvas_process_alive(...) has returned true.